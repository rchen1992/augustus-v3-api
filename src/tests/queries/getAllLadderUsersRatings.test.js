const mockLadderRepo = require('@tests/mockRepos/ladder');
const mockLadderUserRepo = require('@tests/mockRepos/ladderUser');
const createLadderService = require('@services/ladderService');
const { testQuery } = require('@tests/utils');

testQuery({
    id: 'Get user ratings for all users in a ladder',
    query: `
        {
            ladder(id: 1) {
                id
                users {
                    id
                    userName
                    rating
                    ratingDelta
                    rank
                    ladderJoinDate
                }
            }
        }
    `,
    variables: {},
    context: () => {
        return {
            services: {
                ladder: createLadderService(mockLadderRepo, mockLadderUserRepo),
            },
            repos: {
                ladder: mockLadderRepo,
            },
        };
    },
    expected: undefined,
});
