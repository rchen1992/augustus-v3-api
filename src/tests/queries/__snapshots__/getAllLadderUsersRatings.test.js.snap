// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Get user ratings for all users in a ladder 1`] = `
Object {
  "data": Object {
    "ladder": Object {
      "id": "1",
      "users": Array [
        Object {
          "id": "1",
          "ladderJoinDate": "2018-12-11 09:39:31",
          "rank": 1,
          "rating": 160,
          "ratingDelta": 30,
          "userName": "cblanko0",
        },
        Object {
          "id": "2",
          "ladderJoinDate": "2018-09-27 20:12:13",
          "rank": 4,
          "rating": 70,
          "ratingDelta": 30,
          "userName": "csaye1",
        },
        Object {
          "id": "3",
          "ladderJoinDate": "2018-09-27 20:12:13",
          "rank": 3,
          "rating": 110,
          "ratingDelta": -20,
          "userName": "cburchell2",
        },
        Object {
          "id": "5",
          "ladderJoinDate": "2018-09-27 20:12:13",
          "rank": 2,
          "rating": 140,
          "ratingDelta": 40,
          "userName": "selcott4",
        },
      ],
    },
  },
}
`;
